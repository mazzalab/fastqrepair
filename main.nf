#!/usr/bin/env nextflow
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    nf-core/fastqrepair
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Github : https://github.com/nf-core/fastqrepair
    Website: https://nf-co.re/fastqrepair
    Slack  : https://nfcore.slack.com/channels/fastqrepair
----------------------------------------------------------------------------------------
*/

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    IMPORT FUNCTIONS / MODULES / SUBWORKFLOWS / WORKFLOWS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

include { FASTQREPAIR             } from './workflows/fastqrepair'
include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nfcore_fastqrepair_pipeline'
include { PIPELINE_COMPLETION     } from './subworkflows/local/utils_nfcore_fastqrepair_pipeline'
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    NAMED WORKFLOWS FOR PIPELINE
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//
// WORKFLOW: Run main analysis pipeline depending on type of input
//
workflow NFCORE_FASTQREPAIR {

    take:
    samplesheet // channel: samplesheet read in from --input

    main:

    //
    // WORKFLOW: Run pipeline
    //
    FASTQREPAIR (
        samplesheet
    )
    emit:
    multiqc_report = FASTQREPAIR.out.multiqc_report // channel: /path/to/multiqc_report.html
}
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    RUN MAIN WORKFLOW
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

workflow {

    main:
    //
    // SUBWORKFLOW: Run initialisation tasks
    //
    PIPELINE_INITIALISATION (
        params.version,
        params.validate_params,
        params.monochrome_logs,
        args,
        params.outdir,
        params.input
    )

    //
    // WORKFLOW: Run main workflow
    //
    NFCORE_FASTQREPAIR (
        PIPELINE_INITIALISATION.out.samplesheet
    )
    //
    // SUBWORKFLOW: Run completion tasks
    //
    PIPELINE_COMPLETION (
        params.email,
        params.email_on_fail,
        params.plaintext_email,
        params.outdir,
        params.monochrome_logs,
        params.hook_url,
        NFCORE_FASTQREPAIR.out.multiqc_report
    )
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    THE END
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/
