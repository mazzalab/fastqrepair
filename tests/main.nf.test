nextflow_pipeline {

    name "Test fastqrepair pipeline"
    script "../main.nf"

    tag "pipeline"
    tag "fastqrepair"

    test("Run fastqrepair on 30 paired-end reads with profile test") {

        when {
            params {
                // input                   = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastqrepair/testdata/samplesheet_30reads.csv'
                input                   = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastqrepair/testdata/samplesheet_bigdatazoo.csv'
                outdir                  = "$outputDir/30reads"

                publish_dir_all_tools   = true
                num_splits              = 2

                max_cpus                = 2
                max_memory              = '1.GB'
                max_time                = '1.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                // { assert snapshot(
                //     // path("$outputDir/30reads/wiper").list(),
                //     path("$outputDir/30reads/gather/test_30reads_R1.fastq_recovered_merged_report.txt"),
                //     path("$outputDir/30reads/gather/test_30reads_R2.fastq_recovered_merged_report.txt"),
                //     ).match("30reads_paired_end")
                // },
                // { assert new File("$outputDir/30reads/pipeline_info/nf-core_fastqrepair_versions.yml").exists() },
                // { assert new File("$outputDir/30reads/pickup/test_30reads_R1_repaired.fastq.gz").exists() },
                // { assert new File("$outputDir/30reads/pickup/test_30reads_R2_repaired.fastq.gz").exists() },
                // { assert new File("$outputDir/30reads/gather/test_30reads_R1.fastq_recovered_merged_report.txt").readLines()[0].contains("FASTQWIPER SUMMARY:") },
                // { assert new File("$outputDir/30reads/gather/test_30reads_R1.fastq_recovered_merged_report.txt").readLines()[2].contains("68/149 (45.64%)") }

                // { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("single_end_software_versions") }
            )
        }
    }
}
