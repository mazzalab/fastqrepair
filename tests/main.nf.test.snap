{
    "30reads_paired_end": {
        "content": [
            "test_30reads_R1.fastq_recovered_merged_report.txt:md5,0c24cc9a44a41cd022f21c474913ddd3",
            "test_30reads_R2.fastq_recovered_merged_report.txt:md5,53f7d37cd1d554872fe459c332e41025"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-19T16:57:37.198807"
    }
}